File: DevExpress-Demo/VB/MVVMDemo.Wpf/Modules/Commands/POCOCommandsViewModel.vb
Imports System.Windows

Namespace MVVMDemo.Commands

    Public Class POCOCommandsViewModel

        Public Overridable Property CanExecuteSaveCommand As Boolean

        Public Sub Save()
            MessageBox.Show("Action: Save")
        End Sub

        Public Function CanSave() As Boolean
            Return CanExecuteSaveCommand
        End Function

        Public Overridable Property FileName As String

        Public Sub Open(ByVal fileName As String)
            MessageBox.Show(String.Format("Action: Open {0}", fileName))
        End Sub

        Public Function CanOpen(ByVal fileName As String) As Boolean
            Return Not String.IsNullOrEmpty(fileName)
        End Function

        Protected Sub New()
            CanExecuteSaveCommand = True
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/MVVMDemo.Wpf/Modules/Commands/POCOCommandsViewModel.cs
using System.Windows;

namespace MVVMDemo.Commands {
    public class POCOCommandsViewModel {
        public virtual bool CanExecuteSaveCommand { get; set; }
        public void Save() {
            MessageBox.Show("Action: Save");
        }
        public bool CanSave() {
            return CanExecuteSaveCommand;
        }

        public virtual string FileName { get; set; }
        public void Open(string fileName) {
            MessageBox.Show(string.Format("Action: Open {0}", fileName));
        }
        public bool CanOpen(string fileName) {
            return !string.IsNullOrEmpty(fileName);
        }

        protected POCOCommandsViewModel() {
            CanExecuteSaveCommand = true;
        }
    }
}
